common/src/main/java/com/microsoft/identity/common/components/AndroidPlatformComponentsFactory.java

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.components;
+
+import android.app.Activity;
+import android.content.Context;
+
+import androidx.annotation.Nullable;
+import androidx.fragment.app.Fragment;
+
+import com.microsoft.identity.common.crypto.AndroidAuthSdkStorageEncryptionManager;
+import com.microsoft.identity.common.internal.net.cache.HttpCache;
+import com.microsoft.identity.common.internal.platform.AndroidBroadcaster;
+import com.microsoft.identity.common.internal.platform.AndroidDeviceMetadata;
+import com.microsoft.identity.common.internal.platform.AndroidPlatformUtil;
+import com.microsoft.identity.common.internal.providers.oauth2.AndroidTaskStateGenerator;
+import com.microsoft.identity.common.internal.ui.AndroidAuthorizationStrategyFactory;
+import com.microsoft.identity.common.java.WarningType;
+import com.microsoft.identity.common.java.interfaces.IPlatformComponents;
+import com.microsoft.identity.common.java.interfaces.PlatformComponents;
+import com.microsoft.identity.common.java.net.DefaultHttpClientWrapper;
+import com.microsoft.identity.common.java.platform.Device;
+import com.microsoft.identity.common.logging.Logger;
+
+import lombok.NonNull;
+
+/**
+ * A factory class for building Android implementations of platform-dependent components in Common.
+ */
+public class AndroidPlatformComponentsFactory {
+    /**
+     * True if all of the platform-dependent static classes have been initialized.
+     */
+    private static boolean sGlobalStateInitalized = false;
+
+    /**
+     * Initializes platform-dependent static classes.
+     */
+    public static synchronized void initializeGlobalStates(@NonNull final Context context){
```

Comment:
private?
unless we want to initialize these without creating a AndroidPlatformComponent

Reply:
This and fillBuilder is required by AndroidBrokerPlatformComponents.

(well... technically not required, but I would prefer not duplicating the code).

Reply:
protected?

Reply:
AndroidBrokerPlatformComponentFactory does not inherit from AndroidPlatformComponentFactory.

(tried. doesn't make much sense since they don't share the exact same factory method signatures)